File: app/src/main/java/com/recursia/popularmovies/data/mappers/EntityToGenreDatabaseModelMapper.java
package com.recursia.popularmovies.data.mappers;

import com.recursia.popularmovies.data.models.GenreDatabaseModel;
import com.recursia.popularmovies.domain.models.Genre;
import com.recursia.popularmovies.utils.Mapper;

public class EntityToGenreDatabaseModelMapper extends Mapper<Genre, GenreDatabaseModel> {
    @Override
    public GenreDatabaseModel transform(Genre genre) {
        GenreDatabaseModel genreDatabaseModel = new GenreDatabaseModel();
        genreDatabaseModel.setId(genre.getId());
        genreDatabaseModel.setMovieId(genre.getMovieId());
        genreDatabaseModel.setName(genre.getName());
        return genreDatabaseModel;
    }
}

File: app/src/main/java/com/recursia/popularmovies/data/mappers/EntityToGenreDatabaseModelMapper.kt
package com.recursia.popularmovies.data.mappers

import com.recursia.popularmovies.data.models.GenreDatabaseModel
import com.recursia.popularmovies.domain.models.Genre
import com.recursia.popularmovies.utils.Mapper

class EntityToGenreDatabaseModelMapper : Mapper<Genre, GenreDatabaseModel>() {
    override fun transform(genre: Genre): GenreDatabaseModel {
        val genreDatabaseModel = GenreDatabaseModel()
        genreDatabaseModel.id = genre.id
        genreDatabaseModel.movieId = genre.movieId
        genreDatabaseModel.name = genre.name
        return genreDatabaseModel
    }
}
